File: 2020-2021 ZIMA/Zaawansowany c++/lista6/zad3.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <unordered_map>
using namespace std;

auto freqElems(vector<int>& x)
{
    unordered_map<int, int> counters;
    for (auto i : x)
    {
        ++counters[i];
    }

    vector<pair<int, int>> asda;  
    int maxCounter = (*max_element(counters.begin(), counters.end(), 
                                [] (pair<int, int> a, pair<int, int> b) {
                                    return a.second < b.second; })).second;

    for_each(counters.begin(), counters.end(), [&] (pair<int, int> a) { if(a.second == maxCounter) { asda.push_back(a); }}); 
    return asda;
}

int main()
{
    vector<int> test =  {1, 1, 3, 5, 8, 9, 5, 8, 8, 5};
    for(auto x : freqElems(test))
    {
        cout << "(" << x.first << "," << x.second << ")" << endl;
    }
}

File: Software Development/l3/zad3.js
function forEach(a, f){
    for(let i = 0; i < a.length; i++){
        f(a[i]);
    }
}

a = [1, 3];
forEach(a, ( _ ) => { return _ + 1;});
function fe(n){ console.log( n ); };

forEach(a, fe);

function map(a, f){
    for(let i = 0; i < a.length; i++){
        a[i] = f(a[i]);
    }
}

function filter(a, f){
    pom = [];
    for(let i = 0; i < a.length; i++){
        if(f(a[i]))
            pom.push(a[i]);
    }
    return pom;
}

a = [1,2,3,4];
var fil_pom = function (n){
    return n < 3;
}

console.log(filter( a, _ => _ < 3 ));
console.log(filter( a, fil_pom));

map( a, _ => _ * 2 );
console.log(a);
function asda(n){
    return n * 2;
}
map( a, asda);
console.log(a);
